src/components/Table/components/ExtendedColumnItem/ExtendedColumnItem.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import useTheme from 'hooks/useTheme';
+import { ExtendedColumn, Sort } from '../../types';
+import { containerStyles, contentStyles } from './ExtendedColumnItem.style';
+import Icon from 'components/Icon';
+import Tooltip from 'components/Tooltip';
+import { isItemString } from '../../utils/utils';
+
+type Props = {
+  item: ExtendedColumn | string;
+  isNumerical?: boolean;
+  sorting?: Sort;
+};
+
+const ExtendedColumnItem: React.FC<Props> = ({ item, sorting, isNumerical }) => {
+  const theme = useTheme();
+
+  const sortingItem = () =>
+    //TODO: Remove type check when backwards-compatibility is removed
+    !isItemString(item) &&
+    item?.isSortable &&
+    (item.content === sorting?.column ? (
+      <div
+        key={`table_icon_sort_${item.content.toLowerCase()}`}
```

Comment:
minor note: you can create a const on top for `item.content.toLowerCase()` e.g

`const lowerCaseContent = item.content.toLowerCase()` and use this const because i see 6 times this function call in this file